dbms/src/Server/Client.cpp

```diff
@@ -197,7 +199,26 @@ class Client : public Poco::Util::Application
 
             default_database = config.getString("database", "");
             user = config.getString("user", "");
-            password = config.getString("password", "");
+
+            if (config.getBool("ask-password", false))
+            {
+                if (config.has("password"))
+                    throw Exception("Specified both --password and --ask-password. Remove one of them", ErrorCodes::BAD_ARGUMENTS);
+
+                std::cout << "Password for user " << user << ": ";
+                if (!SetStdinEcho(false))
+                    throw Exception("Can't disable input echo", ErrorCodes::UNKNOWN_EXCEPTION);
```

Comment:
The function `setStdinEcho` should throw by itself.

Reply:
We should add an integration test (using `expect`) to check that it can successfully accept a password with spaces.